packages/workspace-resolver/src/resolve-workspace.ts

```diff
@@ -0,0 +1,187 @@
+import path, { join } from 'path';
+import { readJson } from 'fs-extra';
+import globby from 'globby';
+import semver from 'semver';
+
+import type {
+  ModularWorkspacePackage,
+  ModularType,
+  WorkspaceMap,
+  WorkspaceObj,
+} from 'modular-types';
+
+function packageJsonPath(dir: string) {
+  return dir.endsWith('package.json') ? dir : join(dir, 'package.json');
+}
+
+function resolveWorkspacesDefinition(
+  cwd: string,
+  def: PackageJson['workspaces'],
+): string[] {
+  if (!def) {
+    return [];
+  }
+
+  if (Array.isArray(def)) {
+    return def.flatMap((path: string) => {
+      return globby.sync(
+        [`${path}/package.json`, '!**/node_modules/**/*', '!**/__tests__/**/*'],
+        {
+          absolute: false,
+          cwd,
+        },
+      );
+    });
+  }
+
+  return resolveWorkspacesDefinition(cwd, def.packages);
+}
+
+type PackageJson = {
```

Comment:
We normally use [these definitions](https://www.npmjs.com/package/@schemastore/package) through the whole project. If you need something that's not in there, could you extend one of their types?